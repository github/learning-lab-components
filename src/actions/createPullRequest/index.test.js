const createPullRequest = require('./index')
import mockContext from '../../tests/mockContext'

describe('createPullRequest', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      pullRequests: {
        create: jest.fn(() => Promise.resolve({ data: { number: 1 } }))
      },
      issues: {
        createComment: jest.fn()
      }
    })
  })

  it('opens a pull request', async () => {
    await createPullRequest(context, { title: 'My pull request', body: 'a-body.md', head: 'branch' })
    expect(context.github.pullRequests.create).toHaveBeenCalled()
    expect(context.github.pullRequests.create.mock.calls).toMatchSnapshot()
  })

  it('opens a pull request with comments', async () => {
    await createPullRequest(context, { title: 'My pull request', body: 'a-body.md', head: 'branch', comments: ['a-comment.md', 'another-comment.md'] })
    expect(context.github.pullRequests.create).toHaveBeenCalled()
    expect(context.github.issues.createComment).toHaveBeenCalledTimes(2)
    expect(context.github.issues.createComment.mock.calls).toMatchSnapshot()
  })
})
