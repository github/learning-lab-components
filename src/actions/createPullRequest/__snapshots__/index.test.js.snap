// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPullRequest opens a pull request 1`] = `
Array [
  Array [
    Object {
      "base": "master",
      "body": "a-body.md",
      "head": "branch",
      "owner": "JasonEtco",
      "repo": "example",
      "title": "My pull request",
    },
  ],
]
`;

exports[`createPullRequest opens a pull request with comments 1`] = `
Array [
  Array [
    Object {
      "body": "a-comment.md",
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
  Array [
    Object {
      "body": "another-comment.md",
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;
